cardinal/world.go

```diff
@@ -207,6 +208,41 @@ func NewMockWorld(opts ...WorldOption) (*World, error) {
 	return world, nil
 }
 
+func RegisterMessage[In any, Out any](world *World, name string, opts ...message.MessageOption[In, Out]) error {
+	msgType := message.NewMessageType[In, Out](name, opts...)
+	var msg types.Message = msgType
+	err := world.registerMessagesByName(msg)
+	if err != nil {
+		return eris.Wrap(err, "failed to register message")
+	}
+	var ok bool
+	msg, ok = world.GetMessageManager().GetMessageByName(name)
+	if !ok {
+		return eris.Errorf("Failed to register message with name: %s", name)
+	}
```

Comment:
this returns an error for the case where this message does not exist yet. needs to be the other way around

also, we should provide more context in the error message. i.e. `this message has already been registered`

Reply:
I took this out. This logic was redundant.

Reply:
I'm reviewing old PRs to find out issues and complaints. 

@technicallyty In this review, You are incorrect. The logic here is 100% viable and you are making an incorrect assumption here. If you look above the code the message is indeed registered by name. Then in the subsequent code it grabs the same message that was registered. 

This was just defensive coding in case the registerByMessageName function was mutating the message. But since it's passed by reference it's redundant logic as side effects from register will effect the original msg regardless.